File: src/main/kotlin/c_compound_conditions/exercise_33.kt
package c_compound_conditions

// Write a program to approve or deny a bank loan to buy a house. The program will ask for the house value, the buyer's
// salary, and in how many years he/she will pay. Calculate the monthly payment value, knowing that it cannot exceed
// 30% of the salary, otherwise the loan will be denied.

import kotlin.math.pow
import kotlin.math.round

fun bankLoan() {
    print("Home price: ")
    val homePrice = readln().toDouble()

    print("Buyer's Salary: ")
    val buyerSalary = readln().toDouble()
    val _buyerSalary = String.format("%.2f", buyerSalary)

    print("Payment Years: ")
    val yearsPayment = readln().toDouble()

    val monthlyPayment = homePrice / (yearsPayment * 12)
    val _monthlyPayment = round(monthlyPayment * 10.0.pow(2)) / 10.0.pow(2)

    val percentage = 30
    val loanLimit = (buyerSalary * percentage) / 100
    val _loanLimit = String.format("%.2f", loanLimit)

    if (monthlyPayment <= loanLimit) {
        println("------------- APPROVED -------------")
        println("Limit: R$$_loanLimit")
        println("Salary: R$$_buyerSalary")
        println("Month Payment: R$$_monthlyPayment")
        println("----------------------------------------")
    }

    else {
        println("------------- REFUSED -------------")
        println("Limit: R$$_loanLimit")
        println("Salary: R$$_buyerSalary")
        println("Month Payment: R$$_monthlyPayment")
        println("----------------------------------------")
    }
}

File: Dart/c_compound_conditions/exercise_33.dart
// Write a program to approve or deny a bank loan to buy a house. The program will ask for the house value, the buyer's
// salary, and in how many years he/she will pay. Calculate the monthly payment value, knowing that it cannot exceed
// 30% of the salary, otherwise the loan will be denied.

import 'dart:io';

const percentage = 30;

void main() {
  double homePrice = readDouble('home price: ');
  double buyerSalary = readDouble('buyer\'s salary: ');
  double yearsPayment = readDouble('payment years: ');

  double monthlyPayment = homePrice / (yearsPayment * 12);
  double loanLimit = (buyerSalary * percentage) / 100;

  if (monthlyPayment <= loanLimit) {
    print('------------- APPROVED -------------');
    print('Limit: R\$$loanLimit');
    print('Salary: R\$$buyerSalary');
    print('Month Payment: R\$$monthlyPayment');
    print('----------------------------------------');
  } else {
    print('------------- REFUSED -------------');
    print('Limit: R\$$loanLimit');
    print('Salary: R\$$buyerSalary');
    print('Month Payment: R\$$monthlyPayment');
    print('----------------------------------------');
  }
}

double readDouble(String prompt) {
  double? value;

  while (true) {
    stdout.write(prompt);
    String? input = stdin.readLineSync();
    value = double.tryParse(input!);

    if (value != null && value >= 0) {
      return value;
    } else {
      print('Error: Please enter a valid number.');
    }
  }
}
